src/app/inventory/store/d2-item-factory.service.ts

```diff
@@ -648,7 +649,28 @@ function getClassTypeNameLocalized(defs: D2ManifestDefinitions, type: DestinyCla
     return t('Loadouts.Any');
   }
 }
+function getSeason(item) {
+  let sourceToSeason;
+  if (item.classified) {
+    return D2CalculatedSeason;
+  }
+  if (
+    D2SeasonToSource.categoryBlacklist.filter((itemHash) =>
+      item.itemCategoryHashes.includes(itemHash)
+    ).length ||
+    !item.itemCategoryHashes.length ||
+    item.typeName === 'Unknown'
+  ) {
+    return -1;
+  }
+  Object.keys(D2SeasonToSource.seasons).forEach((season) => {
+    if (D2SeasonToSource.seasons[season].includes(item.source)) {
```

Comment:
Would you mind documenting the `source` property (and the other properties like it) in `item-types.ts`? I'm not actually super sure what "source" means and so it's a bit hard to follow what the mappings in `D2SeasonToSource` are.